src/test/java/edu/ucsb/ucsbcslas/controllers/RoomSlotControllerTests.java

```diff
@@ -0,0 +1,193 @@
+package edu.ucsb.ucsbcslas.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.lang.IllegalArgumentException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import static org.junit.Assert.fail;
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.ucsb.ucsbcslas.advice.AuthControllerAdvice;
+import edu.ucsb.ucsbcslas.entities.RoomSlot;
+import edu.ucsb.ucsbcslas.repositories.RoomSlotRepository;
+import edu.ucsb.ucsbcslas.entities.ActiveQuarter;
+import edu.ucsb.ucsbcslas.repositories.ActiveQuarterRepository;
+import edu.ucsb.ucsbcslas.models.Course;
+import jdk.jfr.Timestamp;
+
+import java.time.LocalTime;
+import java.time.DayOfWeek;
+
+@WebMvcTest(value = RoomSlotController.class)
+@WithMockUser
+public class RoomSlotControllerTests {
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @MockBean
+  AuthControllerAdvice mockAuthControllerAdvice;
+
+  @MockBean
+  RoomSlotRepository mockRoomSlotRepository;
+  
+  private String userToken() {
+      return "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTYiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1MTYyMzkwMjJ9.MkiS50WhvOFwrwxQzd5Kp3VzkQUZhvex3kQv-CLeS3M";
+    }
```

Comment:
You may want to comment this noting that isn't a real token. I don't know about you, but I personally can't do a base64 decode in my head to know that this is a dummy token :)